rust_snuba/src/processors/replays.rs

```diff
@@ -144,14 +155,14 @@ pub fn deserialize_message(
                 retention_days: replay_message.retention_days,
                 sdk_name: event.sdk.name.unwrap_or_default(),
                 sdk_version: event.sdk.version.unwrap_or_default(),
-                segment_id: event.segment_id,
                 timestamp: event.timestamp as u32,
                 trace_ids: event.trace_ids.unwrap_or_default(),
                 urls: event.urls.unwrap_or_default(),
                 user,
                 user_email: event.user.email.unwrap_or_default(),
                 user_id: user_id.unwrap_or_default(),
                 user_name: event.user.username.unwrap_or_default(),
+                segment_id,
```

Comment:
There's always a Rust 1-liner in case you need it.
```suggestion
+                segment_id: event.segment_id.filter(|s| *s <= (u16::MAX as u32)).map_or(None, |s| Some(s as u16)),
```